optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java

```diff
@@ -221,19 +228,25 @@ protected EmployeeRosterView getEmployeeRosterView(Integer tenantId, LocalDate s
 
         employeeRosterView.setEmployeeList(employeeList);
 
-        // TODO use startDate and endDate in query
-        List<Shift> shiftList = entityManager.createNamedQuery("Shift.findAll", Shift.class)
-                .setParameter("tenantId", tenantId)
-                .getResultList();
         Map<Long, List<ShiftView>> employeeIdToShiftViewListMap = new LinkedHashMap<>(employeeList.size());
         List<ShiftView> unassignedShiftViewList = new ArrayList<>();
         Set<Employee> employeeSet = new HashSet<>(employeeList);
+        TenantConfiguration tenantConfig = entityManager.createNamedQuery("TenantConfiguration.find",
+                TenantConfiguration.class)
+                .setParameter("tenantId", tenantId).getSingleResult();
+
+        List<Shift> shiftList = entityManager.createNamedQuery("Shift.filterWithEmployees", Shift.class)
+                .setParameter("tenantId", tenantId)
+                .setParameter("startDateTime", startDate.atStartOfDay(tenantConfig.getTimeZone()).toOffsetDateTime())
+                .setParameter("endDateTime", endDate.atStartOfDay(tenantConfig.getTimeZone()).toOffsetDateTime())
+                .setParameter("employees", employeeSet)
```

Comment:
Can we use the same variable name in the JPA-QL query as we do in the rest code? So rename "employees" to "employeeSet".

Reply:
Fixed